shared/eip712/eip712.go

```diff
@@ -0,0 +1,69 @@
+package eip712
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type EIP712Components struct {
+	V uint8    `json:"v"`
+	R [32]byte `json:"r"`
+	S [32]byte `json:"s"`
+}
+
+const EIP712Length = 65
+
+// Pretty print for EIP712Components
+func (e EIP712Components) String() {
+	fmt.Printf("EIP712 Components:\n")
+	fmt.Printf("V: %d\n", e.V)
+	fmt.Printf("R: %x\n", e.R)
+	fmt.Printf("S: %x\n", e.S)
+}
+
+// SanitizeHexInput checks for and strips a 0x prefix, returning an error if one is not found.
+// It returns the byte slice representing the decoded hex, or a decoding error
+func SanitizeHex(inp string) ([]byte, error) {
+	if !strings.HasPrefix(inp, "0x") {
+		return nil, errors.New("expected hex prefix")
+	}
+	return hex.DecodeString(strings.TrimPrefix(inp, "0x"))
+}
+
+// SanitizeEIP712String checks that a 0x-prefixed hex string is the proper length to be
+// parsed as a EIP-712 signature, and parses it, returning an error if the length is invalid
+// or the hex cannot be decoded.
+func SanitizeEIP712String(inp string) ([]byte, error) {
```

Comment:
Can we instead use a function with the signature:
`func (out *EIP712String) Decode(inp string) error`
This would replace SanitizeEIP712String and ParseEIP712Components

If we just want to check that a string is valid we can discard the local variable afterwards.

If we actually want to parse and use something, it lets us write this code:
```golang
eip712Components := new(EIP712Components)
err := eip712Components.Decode(inp)
if err != nil {
    return err
}

err = eip712Components.ValidateSignature(msg string)
if err != nil {
    return err
}
```

Which is succinct and go-like.
You will want to implement `Validate()` in this package, as well as `Encode() string` so that you can round-trip a string for testing/printing/logging